### PR TITLE
VNext Nuget feed

### PR DESCRIPTION
**Pull request by nulltoken**
Where can i find the Vnext nuget feed for the library

### CODE REVIEW

**Comment by nulltoken** [comment]
Ok. I've started to work on this and transmuted the issue into a PR.

**Comment by nulltoken** [comment]
**Next steps:**

 - [ ] Change the CI/build.msbuild so that `Deploy`depends on `Build`
 - [ ] Upon a push to `vNext` or `master`, once all the jobs succeeds, trigger a new "clean" build without the `/property:ExtraDefine="LEAKS_IDENTIFYING"` (a special mode witch ensure we properly release unmanaged handles, otherwise leading to failed tests).
 - [ ] Create the NuGet package by leveveraging **[build.nuget.package.cmd](https://github.com/libgit2/libgit2sharp/blob/vNext/nuget.package/build.nuget.package.cmd)**
 - [ ] Drop the NativeBinaries folder from the symbols package
 - [ ] Publish as artifacts

**Later, when thoroughly tested:**

 - [ ] Add the API key as a secure variable
 - [ ] Publish NuGet package from vNext as `preRelease`
 - [ ] Publish NuGet package from master as `stable`

File: appveyor.yml [hunk 1]
```diff
@@ -24,6 +36,10 @@ test_script:
 - '%xunit_runner%  "C:\projects\libgit2sharp\LibGit2Sharp.Tests\bin\Release\LibGit2Sharp.Tests.dll" /appveyor'
 - IF %ERRORLEVEL% NEQ 0 EXIT /B %ERRORLEVEL%
 
+on_success:
+- ps: |
+    "C:\projects\libgit2sharp\nuget.package\BuildNugetPackage.ps1" "{$env:APPVEYOR_REPO_COMMIT}"
```
Comment (nulltoken) [inline appveyor.yml]: @FeodorFitsner Help :pray: I can't seem to find the proper way to pass the `APPVEYOR_REPO_COMMIT` as a parameter to a `.ps1` in the `on_success` event..

**Note:** If you want a good laugh, you can also peek at the pile of fixup commits in this PR to contemplate all my failed attempts :crying_cat_face:

Comment (FeodorFitsner) [inline appveyor.yml]: Seems like you also need a dot in the start, right?

Try:

    . "C:\projects\libgit2sharp\nuget.package\BuildNugetPackage.ps1" $env:APPVEYOR_REPO_COMMIT

Comment (nulltoken) [inline appveyor.yml]: @FeodorFitsner :heart:

File: appveyor.yml [hunk 2]
```diff
@@ -16,14 +16,58 @@ environment:
...
+    [Environment]::SetEnvironmentVariable("SHOULD_BUILD_NUGET", $ShouldBuildNuget , "User")
+    Write-Host "Should build Nuget = $($env:SHOULD_BUILD_NUGET)"
+    Write-Host "Should build Nuget = $([Environment]::GetEnvironmentVariable("SHOULD_BUILD_NUGET","User"))"
+
+assembly_info:
+  patch: true
+  file: LibGit2Sharp\Properties\AssemblyInfo.cs
+  assembly_version: '{version}'
+  assembly_file_version: '{version}'
+  assembly_informational_version: $($([Environment]::GetEnvironmentVariable("ASSEMBLY_INFORMATIONAL_VERSION", "User")))
```
Comment (nulltoken) [inline appveyor.yml]: @FeodorFitsner I may have (another :blush:) problem here. This generates

```
[assembly: AssemblyInformationalVersion("))")]
```

Any idea? You can check the build log to get a better view at what's going on.

Comment (nulltoken) [inline appveyor.yml]: Hmmm. I may have found a workaround. Switching from `init` to `install` seems to work. However, I think there might be something fishy in the pipeline (as shown above).

**Comment by nulltoken** [comment]
I think this is now ready for review.

The version used in the NuGet packages will be of the following formats

 - `0.20.1.{build}` when generated from `master`
 - `0.20.1.{build}-pre` when generated from `vNext`

Although this is not proper SemVer (4 segments instead of 3), Some local testing seem to show that NuGet can accomodate it.

A sample of the published NuGet packages can be found at 
https://ci.appveyor.com/project/libgit2/libgit2sharp/build/0.20.1.86/job/cf0gf1ismcp4eaik/artifacts

**Comment by nulltoken** [comment]
@davidebbo @davidfowl Regarding the comment above, how much is it dumb to rely on this? I know this is not semver compliant, but will NuGet still cope it for some time?

**Note:** When generating the package the command line issues the following:

``` 
Successfully created package 'C:\projects\libgit2sharp\LibGit2Sharp\LibGit2Sharp.0.20.1.86-pre.nupkg'.

WARNING: 1 issue(s) found with package 'LibGit2Sharp'.
 
Issue: Use semantic versioning
Description: Version "0.20.1.86-pre" does not follow semantic versioning guidelines.
Solution: Update your nuspec file or use the AssemblyInformationalVersion assembly attribute to specify a semantic version as described at http://semver.org. 
```

**Comment by davidebbo** [comment]
@davidfowl is the vnext expert, so I'll let him take this one :)

**Comment by nulltoken** [comment]
See #941 for a better versioning proposal

**Comment by nulltoken** [comment]
@satish860 We've started publishing prerelease NuGet packages. Those are automatically generated by AppVeyor upon push/merge against vNext. They're manually published whenever new features/fixes are delivered.

**Pull request merged**